tests_new/integration_tests/modules/notebooks/global_conftest.py

```diff
@@ -0,0 +1,177 @@
+import logging
+
+import pytest
+
+from integration_tests.client import GqlError
+from integration_tests.modules.notebooks.queries import (
+    create_sagemaker_notebook,
+    get_sagemaker_notebook,
+    delete_sagemaker_notebook,
+    list_sagemaker_notebooks,
+)
+from integration_tests.core.stack.utils import check_stack_ready, check_stack_in_progress, wait_stack_delete_complete
+
+from integration_tests.modules.notebooks.aws_clients import VpcClient
+
+from dataall.base.utils.naming_convention import (
+    NamingConventionService,
+    NamingConventionPattern,
+)
+
+log = logging.getLogger(__name__)
+
+
+def create_notebook(client, group, env_uri, vpc_id, subnet_id, tags=[], name='TestNotebook'):
+    notebook = create_sagemaker_notebook(
+        client=client,
+        name=name,
+        group=group,
+        environmentUri=env_uri,
+        VpcId=vpc_id,
+        SubnetId=subnet_id,
+        tags=tags,
+    )
+    check_stack_ready(
+        client=client,
+        env_uri=env_uri,
+        stack_uri=notebook.stack.stackUri,
+        target_uri=notebook.notebookUri,
+        target_type='notebook',
+    )
+    return get_sagemaker_notebook(client, notebook.notebookUri)
+
+
+def delete_notebook(client, env_uri, notebook):
+    check_stack_ready(
+        client=client,
+        env_uri=env_uri,
+        stack_uri=notebook.stack.stackUri,
+        target_uri=notebook.notebookUri,
+        target_type='notebook',
+    )
+    try:
+        return delete_sagemaker_notebook(client, notebook.notebookUri)
+    except GqlError:
+        log.exception('unexpected error when deleting environment')
+        return False
+
+
+"""
+Session envs persist accross the duration of the whole integ test suite and are meant to make the test suite run faster (env creation takes ~2 mins).
+For this reason they must stay immutable as changes to them will affect the rest of the tests.
+"""
+
+
+@pytest.fixture(scope='session')
+def session_notebook1(client1, group1, session_env1, session_id, session_env1_aws_client):
+    resource_name = 'sessionnotebook1'
+    notebook = None
+    try:
+        vpc_client = VpcClient(session=session_env1_aws_client, region=session_env1['region'])
+        vpc_id = vpc_client.create_vpc(vpc_name=resource_name, cidr='172.31.0.0/26')
+        subnet_id = vpc_client.create_subnet(vpc_id=vpc_id, subnet_name=resource_name, cidr='172.31.0.0/28')
+        notebook = create_notebook(
+            client1,
+            group=group1,
+            env_uri=session_env1['environmentUri'],
+            tags=[session_id],
+            vpc_id=vpc_id,
+            subnet_id=subnet_id,
+        )
+        yield notebook
+    finally:
+        if notebook:
+            delete_notebook(client1, session_env1['environmentUri'], notebook)
+            stack_name = NamingConventionService(
```

Comment:
can't we get the stack_name from `notebook.stack.name`?

Reply:
I wanted to ensure we clean up VPC but need to wait until notebook is deleted 

We likely will have to ensure clean up is orchesteated in some sequential order as we add more modules for testing

Reply:
Clean up order is controlled by the fixture creation order so we probably don't need to do anything explicitly. Did you maybe look into using the `notebook.stack.name`?

Reply:
yes i just changed in latest and removed NamingConventionService